io-engine/src/bdev/nexus/nexus_persistence.rs

```diff
@@ -119,6 +121,21 @@ impl<'n> Nexus<'n> {
                     None => nexus_info.children.push(child_info),
                 }
             }
+            PersistOp::RemoveChild {
+                child_uri,
+            } => {
+                let uuid = NexusChild::uuid(&child_uri)
+                    .expect("Failed to get child UUID.");
+
+                let idx =
+                    nexus_info.children.iter().position(|r| r.uuid == uuid);
+                if idx.is_some() {
+                    let c = nexus_info.children.remove(idx.unwrap());
+                    // Making sure that we didn't remove a healthy child.
+                    // This better be an assert and not debug_assert.
+                    assert!(!c.healthy);
+                }
+            }
```

Comment:
We had an issue before that caused duplicate entries in the info, we should probably make sure we remove all:

```suggestion
                nexus_info.children.retain(|child| child.uuid != uuid); 
            }
```

Reply:
nice, didn't know of this `retain` call. So just to reiterate, this single line `retain` call will retain all entries in the vector except the one intended to be removed?

Reply:
Yes correct.